### PR TITLE
KIALI-2852 Improve general website design

### PR DESCRIPTION
**Pull request by israel-hdez**

### CODE REVIEW

File: themes/kiali/theme.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# theme.toml template for a Hugo theme
+# See https://github.com/gohugoio/hugoThemes#themetoml for an example
+
+name = "Kiali"
+license = "MIT"
+licenselink = "https://github.com/yourname/yourtheme/blob/master/LICENSE.md"
+description = ""
+homepage = "http://example.com/"
```
Comment (pilhuhn) [inline themes/kiali/theme.toml]: Probably needs more work.

**Comment by aljesusg** [comment]
Yes, this need more work, https://deploy-preview-153--kiali-io-draft.netlify.com/documentation/ miss style and images are not scale, This is a hard work , I know it because I proposed so time ago ^^ at the moment that we are overwriting the styles and js we should create a kiali theme and remove the old one so +1 for this.

**Comment by abonas** [comment]
I understand this is a draft, just in case this looks different only on my screen (notice how the background image is cut and replicated)
website

File: themes/kiali/theme.toml (hunk 1 above)
Comment (israel-hdez) [reply themes/kiali/theme.toml]: Yes, this is the scaffolded file. I haven't worked on this file.

**Comment by israel-hdez** [comment]
@abonas Yes, I'm aware of the background image issue. Thanks for reviewing :)

**Comment by israel-hdez** [comment]
@aljesusg Yes, indeed my plan is to remove the _canvas_ theme. Despite the design change, I try to do a smooth transition with the least possible changes to the content sources.

**Comment by israel-hdez** [comment]
@abonas The background issue should be fixed.

File: themes/kiali/layouts/_default/baseof.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  <link rel="stylesheet" href="{{ "css/kiali.css" | relURL }}" />
+
+  <!--
+    Syntax hightlighting: https://highlightjs.org/download/
+    Languages pack: CSS, HTTP, JavaScript, Bash, Makefile, Diff, JSON, HTML/XML, Java, Properties,
+      Shell Session, AsciiDoc, Dockerfile, Typescript, YAML, plaintext.
+  -->
+  <link rel="stylesheet" href="{{ "css/highlight.css" | relURL }}">
+  <script src="{{ "js/highlight.pack.js" | relURL }}"></script>
```
Comment (pilhuhn) [inline themes/kiali/layouts/_default/baseof.html]: I think the script tag should have a type element, that describes the media type - especially when we tighten the content security policy. See e.g. https://github.com/kiali/kiali.io/pull/146/files#diff-8ab16b2cae0b26a574b75e8e5c19e378

**Comment by jshaughn** [comment]
@israel-hdez Probably an unnecessary comment but just an FYI that I know various PRs customized the Canvas theme.  For example, I remember @josejulio made changes to support the proper expand/collapse of videos and images.

File: themes/kiali/theme.toml (hunk 1 above)
Comment (israel-hdez) [reply themes/kiali/theme.toml]: Fixed

File: themes/kiali/layouts/_default/baseof.html (hunk 2 above)
Comment (israel-hdez) [reply themes/kiali/layouts/_default/baseof.html]: Fixed.

**Comment by israel-hdez** [comment]
@cfcosta Deploy your screen army to check how it looks on all resolutions :)

**Comment by israel-hdez** [comment]
@jshaughn I tried to preserve most customizations. Hopefully, you won't notice the missing changes to Canvas (if any) :)

**Comment by cfcosta** [comment]
@israel-hdez ok, testing. I don't have a mac to test right now, but will test on Linux and Windows.

**Comment by cfcosta** [comment]
@israel-hdez 
## Chrome on Linux

With forced window size, and browser chrome:

3440x1440: https://imgur.com/ojpcDdW.png
2560x1440: https://imgur.com/iTBf23q.png
2560x1080: https://imgur.com/JmaGmf4.png
1920x1080: https://imgur.com/H7i1jEE.png
1280x720: https://imgur.com/LlpEHUf.png

Full visible page:

3440x1440: https://imgur.com/RQAW6Jg
2560x1440: https://imgur.com/8Frw2nA
2560x1080: https://imgur.com/KV7JD6e
1920x1080: https://imgur.com/YAyFBhM
1280x720: https://imgur.com/biX5Wfv

**Comment by josejulio** [comment]
> * The FAQ would benefit from a TOC

I think we should remove the text in the "Graph" to have only a TOC in the FAQ section
toc

**Comment by israel-hdez** [comment]
@josejulio Hopefully, all your feedback is fixed.

**Comment by israel-hdez** [comment]
@jshaughn Most of your feedback is fixed.

> * We lost the video on the landing page, is that intentional?  It's not obvious/easy to get to this demo by just going to youtube via the demos link.

I made this comment in the design repo: https://github.com/kiali/kiali-design/issues/127#issuecomment-502821519.
It's intentional that it's no longer on the top. I forgot about it :-/ I'll remember Mary to check if she did the mockup with the video after the hero section.

The least intrusive change I can think of is to place a "View introductory video" button and show it "fullscreen" like the magnifying images in the features page, but keeping the video hidden by default.

> * I think we should promote more the configuration AND validation abilities on the landing page.
>   -- It may be better to just remove the "About Kiali" section and instead add a "Kiali lets you:" section up with "Kiali answers the questions:".
>   -- The demos link could move up with the get started and blogs links.

I think this is feedback for Mary and should go in the design issue https://github.com/kiali/kiali-design/issues/127. I'm not comfortable doing these changes without her approval.

> * The FAQ would benefit from a TOC

I though the TOC is the one at https://5d433216c3f521000810dfaf--kiali-io-draft.netlify.com/faq/. (I removed the text that Josejulio mentioned, so It now looks more like a TOC)

**Comment by jshaughn** [comment]
@israel-hdez OK, yes, we need to follow up on the overview video, Mike spent a lot of time producing it, Mary can figure out where to put it.

As for the FAQ TOC, I don't see it.  I would have expected it to have 3 entries.  It's true that currently I can see all 3 headers on screen given a large-ish browser window, but add one more section and you won't see it at all.  If you guys think the current FAQ page basically *is* the TOC, then I guess we can just keep it that way.

I can add my feedback to the design issue.  I guess I missed that whole design issue discussion while I was away.

**Comment by israel-hdez** [comment]
@jshaughn Oh, I think I  understand. So, is your preference something like this one: https://deploy-preview-153--kiali-io-draft.netlify.com/documentation/ ?

For the "faq page" I tried to keep what the current web site is showing in https://www.kiali.io/faq/. But, indeed, I prefer having only the headers. Else, any introductory text may be missed, like the one at the beginning of the graph faq: https://www.kiali.io/faq/graph/. I can go that way if it is OK for others.

**Comment by jshaughn** [comment]
> So, is your preference something like this one:

Not really :)   I'd just like to see at the top a TOC that looks like:

General
Graph
Distributed Tracing

With all of the existing content below it.  It's just a way to see and jump to the section I care about from the initial page, without having to scroll to see all of the FAQ sections/topics.  It's just my preference, it's not a huge deal.

**Comment by israel-hdez** [comment]
> Not really :) I'd just like to see at the top a TOC that looks like:

OK, I understand better. So, yes, for me the `/faq` page is the TOC because clicking each entry will redirect you to other page with the actual content. I think I prefer the current structure, because IMO the titles don't bring a well idea of what to expect. But it could be improved, I guess, to be more inline with your suggestion.

**Comment by jshaughn** [comment]
>  So, yes, for me the /faq page is the TOC because clicking each entry will redirect you to other page with the actual content

Just to make sure we are absolutely understanding each other, I am not proposing you remove/change any of your existing FAQ page.  The sections and questions/links all stay in place.  I am suggesting only that the FAQ page add a simple TOC to be able to jump, within the page, to the sections (currently 3 sections, probably more in the future).  We can keep it as-is, without a TOC, but I just wanted to ensure we understood each other.

**Comment by israel-hdez** [comment]
@jshaughn I've restored the introductory video.

File: themes/kiali/layouts/partials/masthead.html [hunk 3]
```diff
@@ -1,14 +1,17 @@
   <header>
     <div>
-      <a href="/"><img src="/img/kiali_logo_masthead.png" alt="Kiali Logo"></a>
+      <div>
+        <a href="/"><img src="/img/kiali_logo_masthead.png" alt="Kiali Logo"></a>
+      </div>
+      <nav>
+        <!-- Checkbox is for sandwich menu in mobile screens
```
Comment (josejulio) [inline themes/kiali/layouts/partials/masthead.html]: https://xkcd.com/149/

**Comment by israel-hdez** [comment]
I'm going to push the _Squash and merge_ button.
Other improvements can be done in follow-up PRs.

**Comment by aljesusg** [comment]
Yes please I'm waiting to rebase my PR about versions

**Comment by abonas** [comment]
GREAT WORK!

**Pull request merged**